recipes/Aishell1Mix/separation/scripts/create_aishell1mix_from_metadata.py

```diff
@@ -0,0 +1,534 @@
+    aishell1mix_outdir = os.path.join(aishell1mix_outdir, f"Aishell1Mix{n_src}")
+    # Get the desired frequencies
+    freqs = args.freqs
+    freqs = [freq.lower() for freq in freqs]
+    # Get the desired modes
+    modes = args.modes
+    modes = [mode.lower() for mode in modes]
+    types = args.types
+    types = [t.lower() for t in types]
+    # Get the number of sources
+    create_aishell1mix(
+        aishell1_dir,
+        wham_dir,
+        aishell1mix_outdir,
+        metadata_dir,
+        freqs,
+        n_src,
+        modes,
+        types,
+    )
+
+
+def create_aishell1mix(
+    aishell1_dir, wham_dir, out_dir, metadata_dir, freqs, n_src, modes, types
+):
+    """ Generate sources mixtures and saves them in out_dir"""
+    # Get metadata files
+    md_filename_list = [
+        file for file in os.listdir(metadata_dir) if "info" not in file
+    ]
+    # Create all parts of aishell1mix
+    for md_filename in md_filename_list:
+        csv_path = os.path.join(metadata_dir, md_filename)
+        process_metadata_file(
+            csv_path,
+            freqs,
+            n_src,
+            aishell1_dir,
+            wham_dir,
+            out_dir,
+            modes,
+            types,
+        )
+
+
+def process_metadata_file(
+    csv_path, freqs, n_src, aishell1_dir, wham_dir, out_dir, modes, types
+):
+    """ Process a metadata generation file to create sources and mixtures"""
+    md_file = pd.read_csv(csv_path, engine="python")
+    for freq in freqs:
+        # Get the frequency directory path
+        freq_path = os.path.join(out_dir, "wav" + freq)
+        # Transform freq = "16k" into 16000
+        freq = int(freq.strip("k")) * 1000
+
+        for mode in modes:
+            # Path to the mode directory
+            mode_path = os.path.join(freq_path, mode)
+            # Subset metadata path
+            subset_metadata_path = os.path.join(mode_path, "metadata")
+            os.makedirs(subset_metadata_path, exist_ok=True)
+            # Directory where the mixtures and sources will be stored
+            dir_name = (
+                os.path.basename(csv_path)
+                .replace(f"aishell1mix{n_src}_", "")
+                .replace(".csv", "")
+            )
+            dir_path = os.path.join(mode_path, dir_name)
+            # Create subdir
+            if types == ["mix_clean"]:
+                subdirs = [f"s{i + 1}" for i in range(n_src)] + ["mix_clean"]
+            else:
+                subdirs = (
+                    [f"s{i + 1}" for i in range(n_src)] + types + ["noise"]
+                )
+            # If the files already exist then continue, otherwise remove the dir and regenerate
+            sounds = glob.glob(
+                os.path.join(dir_path, "**/*.wav"), recursive=True
+            )
+            if len(md_file) * len(subdirs) == len(sounds):
+                print(
+                    f"Directory {dir_path} already exist. "
+                    f"Files won't be overwritten"
```

Comment:
Hi @ycemsubakan , I have already checked the files were generated properly here.